lib/graphql/dataloader/flat_dataloader.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module GraphQL
+  class Dataloader
+    class FlatDataloader < Dataloader
+      def initialize(*)
+        # TODO unify the initialization lazies_at_depth
+        @lazies_at_depth ||= Hash.new { |h, k| h[k] = [] }
+        @steps_to_rerun_after_lazy = []
+        @queue = []
+      end
+
+      def run
+        while @queue.any?
+          while (step = @queue.shift)
+            step.call
+          end
+
+          while @lazies_at_depth&.any?
+            smallest_depth = nil
+            @lazies_at_depth.each_key do |depth_key|
+              smallest_depth ||= depth_key
+              if depth_key < smallest_depth
+                smallest_depth = depth_key
+              end
+            end
+
+            if smallest_depth
+              lazies = @lazies_at_depth.delete(smallest_depth)
+              lazies.each(&:value) # resolve these Lazy instances
+            end
+          end
+
+          if @steps_to_rerun_after_lazy.any?
+            @steps_to_rerun_after_lazy.each(&:call)
+            @steps_to_rerun_after_lazy.clear
+          end
+        end
+      end
+
+      def run_isolated
```

Comment:
What, basically, does `run_isolated` do? I remember it's used in context of mutations, so I presume its a serial execution concern? In Cardinal we address serial constraints by splitting up the root into separate execution scopes that run serially, effectively treating it as N separate executions that happen to run in sequence.

Reply:
It's used in a couple of hacky places where, for legacy reasons ([which ones??]), we need a dataloader-enabled code block to return right away, without running any other enqueued work. 

Yeah, it seems like initializing a new dataloader, using it for one thing, then discarding it would do. I'll give that a try sometime.